derive_client/clients/base_client.py

```diff
@@ -64,35 +71,47 @@ def _create_signature_headers(self):
             session_key_or_wallet_private_key=self.signer._private_key,
         )
 
+    @validate_arguments(config=dict(arbitrary_types_allowed=True))
     def __init__(
         self,
-        wallet: str,
+        wallet: Address,
```

Comment:
This validates and type casts input (address to check-summed address, subaccount_id is `str` to `int`)